metals/src/main/scala/scala/meta/internal/metals/StatusBar.scala

```diff
@@ -107,9 +115,34 @@ final class StatusBar(
       showTimer: Boolean = false,
       progress: Option[TaskProgress] = None,
   ): Future[T] = {
-    items.add(Progress(message, value, showTimer, progress))
-    tickIfHidden()
-    value
+
+    if (clientConfig.statusBarState == StatusBarState.Off) {
```

Comment:
Strikes me that it might be a good idea to introduce an abstraction over `StatusBar` called `Progress` or similar which could have 2 impls, one for when statusbar support is detected, the other using the LSP progress stuff

Reply:
I'm not the main contributor of `metals` but maybe we can also check if the client has `window.workDoneProgress` capability. If the client supports, metals can use `progress` instead of the existing `status`.

This will be similar to what [gopls does](https://github.com/golang/tools/blob/master/gopls/internal/progress/progress.go#L92).

Reply:
Checking the capability makes sense, though as the status bar support is richer than native LSP progress notifications I wonder if the order of precedence should be status bar > LSP progress > no-op